src/openrct2-ui/windows/Ride.cpp

```diff
@@ -1601,15 +1601,15 @@ static void window_ride_init_viewport(rct_window* w)
     else if (eax >= ride->num_vehicles && eax < (ride->num_vehicles + ride->num_stations))
     {
         StationIndex stationIndex = STATION_INDEX_NULL;
-        int32_t count = eax - ride->num_vehicles;
-        do
+        int count = eax - ride->num_vehicles;
+        while (count >= 0)
```

Comment:
Rename eax to `viewSelectionIndex`. The overflow that we are trying to avoid is `StationIndex stationIndex = STATION_INDEX_NULL + 1` instead you should make it something like this:
```
for (const auto& station : ride->stations){
    if (station.Start.isNull()){
        continue;
    }
    count--;
    if (count == 0){
       // L1614-1618 auto location = station.GetStart(); ....
    }
}
```